tests/IVIMmodels/unit_tests/test_ivim_fit.py

```diff
@@ -161,14 +62,35 @@ def data_ivim_fit_saved():
 
 @pytest.mark.parametrize("name, bvals, data, algorithm, xfail, kwargs, tolerances", data_ivim_fit_saved())
 def test_ivim_fit_saved(name, bvals, data, algorithm, xfail, kwargs, tolerances, request):
+    global test_results
     if xfail["xfail"]:
         mark = pytest.mark.xfail(reason="xfail", strict=xfail["strict"])
         request.node.add_marker(mark)
     fit = OsipiBase(algorithm=algorithm, **kwargs)
     signal, ratio = signal_helper(data["data"])
+    
     tolerances = tolerances_helper(tolerances, ratio, data["noise"])
     [f_fit, Dp_fit, D_fit] = fit.osipi_fit(signal, bvals)
+    def to_list_if_needed(value):
+        return value.tolist() if isinstance(value, np.ndarray) else value
+    test_result = {
+        "name": name,
+        "algorithm": algorithm,
+        "f_fit": to_list_if_needed(f_fit),
+        "Dp_fit": to_list_if_needed(Dp_fit),
+        "D_fit": to_list_if_needed(D_fit),
+        "f": to_list_if_needed(data['f']),
+        "Dp": to_list_if_needed(data['Dp']),
+        "D": to_list_if_needed(data['D']),
+        "status": "PASSED"
+    }
+    if xfail["xfail"]:
+        test_result['status'] = "XFAILED"
+
+    test_results.append(test_result)
+    with open('test_results_report.json', 'w') as f:
```

Comment:
This is writing the entire json for each test which is surely slowing things down a lot. It's also custom code that's going to need to be considered for every change of the test.

What I suggested last time and I'll suggest again is to look into other ways pytest can do this. A 5 minute google search is showing `pytest --json-report`, `Pytest-harvest`, and possibly pytest fixtures. Have you considered any of these?

Reply:
I did look into pytest-json-report but It gives a complex report with unnecessary information, the only way I see to include our custom values is by logging it into the code, and the final result file is not clean after then.

I'm now using pytest fixture for handling report generation.

Reply:
What about `record_property`? https://docs.pytest.org/en/7.1.x/how-to/output.html

Not sure what you mean about the final result file not being clean, but using what is essentially a global variable isn't great. You essentially created a custom fixture that no other test can use because doing so would corrupt the test logging. A "manual" logging solution like this should be the last resort - it's very hard to maintain and because it operates outside of the testing structure it's hard to keep consistent, e.g. capturing the pass/xfail/fail criteria.

Reply:
I got your point, thanks for explaining.
<img width="765" alt="image" src="https://github.com/user-attachments/assets/38a8d171-b4e8-40a6-9818-e500590faf12">
That's how the report looks like, I used `pytest --json-report` with `record_property` for capturing the data. I'll update the frontend depending on that report and update the PR.

Reply:
Ok, just from that it doesn't look too bad. Seems like the code changes are still incoming?

Reply:
yeah I'm getting weird bugs on the frontend after updating code, I'll give it a final attempt and then push.
<img width="452" alt="image" src="https://github.com/user-attachments/assets/6e7f2792-9d62-47fa-9fb6-622248958fe1">